### DISCUSSION TITLE
How to add external libraries? js_cookie__WEBPACK_IMPORTED_MODULE_0__

### DISCUSSION THREAD

**Discussion by capoia** [GrapesJS/grapesjs#5072] [Q&A]
Hello, I needed to work with cookies and masks in the component I'm building using the grapesjs cli.
More precisely I need to use these libraries inside the "script" property in the type definition.
I'm trying to use the vanilla-masker and js-cookies libraries for this, but when using IMPORT inside the components.js file I get the error: "js_cookie__WEBPACK_IMPORTED_MODULE_0__".
I saw that if I import the packages inside index.js they work correctly.
I even tried to pass it via OPTS, but within the script function these variables are not accessible.
I also try to use require() but dont work too.
![CleanShot 2023-04-26 at 11 01 01](https://user-images.githubusercontent.com/21159928/234599996-13c164de-faca-4f85-bb6c-18c297e62f8c.png)

**Comment by capoia** [comment]
@artf Sorry to bother you, but can you help me with this?

**Comment by artf** [comment]
I'd suggest to follow the guide [here](https://grapesjs.com/docs/modules/Components-js.html#component-related) on how to deal with dependencies in your `script`. The suggested dynamic load of dependencies guaranties that they won't be bundled in the end-user page if not used in the editor.

**Reply by capoia** [reply]
> I'd suggest to follow the guide [here](https://grapesjs.com/docs/modules/Components-js.html#component-related) on how to deal with dependencies in your `script`. The suggested dynamic load of dependencies guaranties that they won't be bundled in the end-user page if not used in the editor.

Thanks a lot for the feedback.
I saw that the guide talks about CDN, but is it possible to use the npm package? performance is vital for my application (I use gtmetrix and it complains a lot about cdn).

**Reply by artf** [reply]
Another option is to use [template related dependencies](https://grapesjs.com/docs/modules/Components-js.html#template-related) but then it's up to you to inject them properly in the output HTML.
